scripts/copy-models/index.js

```diff
@@ -23,34 +57,42 @@ const { models } = yargs(process.argv.slice(2))
     .filter((file) => file.isDirectory())
     .map((dir) => join(models.toString(), dir.name, `smithy/model.json`));
 
+  const observedProtocols = new Set();
+
   for (const smithyModelsFile of smithyModelsFiles) {
     try {
       // Test if file exists.
       await fsPromises.stat(smithyModelsFile);
       // File exists, copy it.
+      const absolutePath = resolve(smithyModelsFile);
+
       try {
-        const fileContent = (await fsPromises.readFile(smithyModelsFile)).toString();
+        const model = require(absolutePath);
+        const sdkId = getSdkId(model).toLowerCase().replace(/\s/g, "-");
+        const protocol = getWireProtocol(model);
 
-        const sdkIdRE = /"sdkId": "([^"]*)"/;
-        const sdkId = fileContent.match(sdkIdRE)[1].toLowerCase().replace(/\s/g, "-");
+        observedProtocols.add(protocol);
 
-        // Copy file.
-        const outputFile = join(OUTPUT_DIR, `${sdkId}.json`);
-        await fsPromises.writeFile(outputFile, fileContent);
+        if (!protocols || protocols.split(",").includes(protocol)) {
+          // Copy file.
+          const outputFile = join(OUTPUT_DIR, `${sdkId}.json`);
+          await fsPromises.writeFile(outputFile, JSON.stringify(model, null, 2));
+          console.log("Copied", outputFile);
+        }
       } catch (e) {
         // Copy failed, log.
-        console.log(smithyModelsFile);
+        console.log("Failed to copy", absolutePath);
         console.log(e.message);
       }
     } catch (e) {
       // File doesn't exist, ignore.
-      console.log(e.message);
+      console.log("File not found", e.message);
     }
   }
 
-  // Prettify copied models
-  await spawnProcess(join(__dirname, "..", "..", "node_modules", ".bin", "prettier"), [
-    "--write",
-    `${OUTPUT_DIR}/*.json`,
-  ]);
```

Comment:
removing prettier step to save time. `JSON.stringify(, null, 2)` is good enough.